test/spark-api.js

```diff
@@ -6,31 +6,70 @@ const recordTelemetry = (measurementName, fn) => { /* no-op */ }
 describe('spark-api client', () => {
   it('fetches round details', async function () {
     this.timeout(10_000)
-    const { retrievalTasks, maxTasksPerNode, ...details } = await fetchRoundDetails(
-      '0x8460766edc62b525fc1fa4d628fc79229dc73031',
-      12600n,
-      recordTelemetry
-    )
-
-    assert.deepStrictEqual(details, {
-      roundId: '18024', // BigInt serialized as String,
-      startEpoch: '4158303'
-    })
-
-    assert.strictEqual(typeof maxTasksPerNode, 'number')
-
-    assert.strictEqual(retrievalTasks.length, 1000)
-    assert.deepStrictEqual(retrievalTasks.slice(0, 2), [
+    const tests = [
       {
-        cid: 'bafkreia3oovvt7sws7wnz43zbr33lsu2yrdmx4mqswdumravjrnxfoxdka',
-        minerId: 'f02228866',
-        clients: ['f01990536']
+        contractAddress: '0x8460766edc62b525fc1fa4d628fc79229dc73031',
+        roundIndex: 12600n,
+        expectedTaskCount: 1000,
+        expectedRoundDetails: {
+          roundId: '18024', // BigInt serialized as String,
+          startEpoch: '4158303'
+        },
+        expectedTasks: [
+          {
+            cid: 'bafkreia3oovvt7sws7wnz43zbr33lsu2yrdmx4mqswdumravjrnxfoxdka',
+            minerId: 'f02228866',
+            clients: ['f01990536'],
+            allocators: null
+          },
+          {
+            cid: 'bafkreibipuscsrko7tlrw62rttqvbma3qqqkksjoi6bhvwp27qaylwupp4',
+            minerId: 'f02982293',
+            clients: ['f03064945'],
+            allocators: null
+          }
+        ]
       },
       {
-        cid: 'bafkreibipuscsrko7tlrw62rttqvbma3qqqkksjoi6bhvwp27qaylwupp4',
-        minerId: 'f02982293',
-        clients: ['f03064945']
+        contractAddress: '0x8460766edc62b525fc1fa4d628fc79229dc73031',
+        roundIndex: 27774n,
+        expectedRoundDetails: {
+          roundId: '33198',
+          startEpoch: '4799802'
+        },
+        expectedTaskCount: 666,
+        expectedTasks: [
+          {
+            cid: 'bafk2bzacea2fqjim35ouwqh4cjlxf7nov4qgc65ufu2nzpxuhv5utdqy54jne',
+            minerId: 'f03233333',
+            clients: ['f03493383'],
+            allocators: ['f03019261']
+          },
+          {
+            cid: 'bafk2bzaceaqybia3guy7szzvgkoshgmxdxknlm4hdyp3mb466tmwq264ky3ec',
+            minerId: 'f03408862',
+            clients: ['f03266237'],
+            allocators: ['f03018489']
+          }
+        ]
       }
-    ])
+    ]
+
+    for (const test of tests) {
+      const { retrievalTasks, maxTasksPerNode, ...details } = await fetchRoundDetails(
```

Comment:
When one of the assertions fails, we won't know which round was tested. When the first loop iteration fails, we won't learn whether the second case passes or fails.

Please rework this into a series of test cases.

Also the name `test` is often used to define new test cases (`import {test} from 'node:test'`), please use a different variable name. How about `tc` for test-case?

```js
for (const tc of tests) {
  it(`fetches round details for round ${tc.roundIndex} of contract ${tc.contractAddress}', async function () {
    this.timeout(10_000)
    // ...
  })
}
```

Reply:
Thanks for feedback! I've refactored the code so every test case runs separately as you've suggested https://github.com/CheckerNetwork/spark-evaluate/pull/502/commits/cd24354fdc1df6c783154baac27ad582e4002ea9. Also `tc` for test-case sounds good to me! 👍🏻